util/truncate/truncate.go

```diff
@@ -8,9 +8,10 @@ package truncate
 // String returns a prefix of a UTF-8 string s, having length no greater than n
 // bytes. If s exceeds this length, it is truncated at a point ≤ n so that the
 // result does not end in a partial UTF-8 encoding.  If s is less than or equal
-// to this length, it is returned unmodified.
+// to this length, it is returned unmodified. If n is negative, s is also returned
+// unmodified.
 func String[String ~string | ~[]byte](s String, n int) String {
-	if n >= len(s) {
+	if n < 0 || n >= len(s) {
```

Comment:
Although I understand the desire to avert a panic, this solution introduces a different odd semantic corner: The rule is supposed to be you wind up with a string no longer than n bytes, but now it may not be truncated at all. 

Personally I'd rather let it panic and reveal what is almost certainly a programming error, but if not it seems like the result in this case should be `""` rather than the whole string, right?